src/tests/test_pubkey.cpp

```diff
@@ -647,6 +648,33 @@ std::vector<Test::Result> PK_Key_Generation_Test::run() {
                            !public_key->supports_operation(Botan::PublicKeyOperation::KeyAgreement));
          }
 
+         // Test that the raw public key can be encoded. This is not supported
+         // by all algorithms; we expect Not_Implemented for these.
+         const std::vector<std::string> algos_that_dont_have_a_raw_encoding = {"RSA"};
+         try {
+            auto raw = public_key->raw_public_key_bits();
+            result.test_ne("raw_public_key_bits is not empty", raw.size(), 0);
+
+            if(public_key->supports_operation(Botan::PublicKeyOperation::KeyAgreement)) {
+               // For KEX algorithms, raw_public_key_bits must be equal to the canonical
+               // public value obtained by PK_Key_Agreement_Key::public_value().
+               const auto* ka_key = dynamic_cast<const Botan::PK_Key_Agreement_Key*>(&key);
+               result.require("is a key agreement private key", ka_key != nullptr);
+               result.test_eq("public_key_bits has same encoding", raw, ka_key->public_value());
```

Comment:
If the dynamic cast fails this will crash with a null deref

Reply:
```
if(const auto* ka_key = dynamic_cast...) {
  result.test_eq("public_key_bits has same encoding", raw, ka_key->public_value());
} else {
  result.test_failure("Not a PK_Key_Agreement_Key");
}
```

Reply:
Ah yes sorry I missed this and thought you were using `confirm` there. All good then.